cpp/src/arrow/python/serialize.cc

```diff
@@ -90,8 +90,9 @@ class SequenceBuilder {
                          MakeBuilderFn make_builder) {
     if (!*child_builder) {
       child_builder->reset(make_builder());
-      // std::to_string is locale dependent, but should be ok for small integers
-      type_map_[tag] = builder_->AppendChild(*child_builder, std::to_string(tag));
+      std::ostringstream convert;
+      convert << static_cast<int>(tag);
```

Comment:
Like above, the `<<` operator is also locale-dependent by default.

One way to avoid this is to call the [imbue](https://en.cppreference.com/w/cpp/io/ios_base/imbue) method to change to the C locale, for example `stream.imbue(std::locale::classic())`. Also note this isn't very fast.